### DISCUSSION TITLE
Request to make validation optional for Stringify OR export unsafeStringify

### DISCUSSION THREAD

**Discussion by vineet-sh** [uuidjs/uuid#663] [Q&A] [answered]
Previously I was using `bytesToUuid`  (v3.4.0) which I understand is now `stringify`, but uuid validation in `stringify` is invalidating most of uuids. It would be great if we can do any of the following:
 - Either export the `unsafeStringify` as a separate API 
 - Or make validation part in `stringify` as optional

**Comment by LinusU** [comment]
> [...] is invalidating most of uuids.

What do you mean by this? 🤔 

Are you trying to stringify a byte array that is not a uuid?

**Reply by vineet-sh** [reply]
Yes, those were unconstrained 128 bit numbers which were converted to uuid like format using `bytesToUuid`.
Presently, `stringify` validates those against uuid regex and invalidates them, which is although the correct behavior, but it would be great if `bytesToUuid` (presently `unsafeStringify`) functionality can be restored along with the present APIs

**Comment by broofa** [comment, marked as answer]
> Either export the unsafeStringify as a separate API

[My rationale for why we shouldn't do this](https://github.com/uuidjs/uuid/pull/597#issuecomment-1003181360).  tl;dr: it increases the maintenance burden.

> Or make validation part in stringify as optional

I appreciate that `bytesToUuid()` used to take any 16-byte array, but `uuid` is no longer in the business of generating or otherwise working with invalid UUIDs.  That it complains loudly and early upon encountering a bad UUID is, to my mind, a good thing.  It allows us to keep the code and public API focused, and helps users quickly identify issues with upstream code.  (Instead of, for example, allowing `stringify()` to generate invalid UUIDs that might propagate into DB tables, URLs, etc.)

**Reply by vineet-sh** [reply]
Got it, thanks :)
